internal/models/questions/questions.go

```diff
@@ -460,7 +480,61 @@ func AddNewQuestion(db *sql.DB, question Question) error {
 
 	// Insert the alternatives into the database
 	for _, a := range question.Alternatives {
-		tx.Exec(
+		_, err = tx.Exec(
+			`INSERT INTO answer_alternatives (id, text, correct, question_id)
+			VALUES ($1, $2, $3, $4);`,
+			a.ID, a.Text, a.IsCorrect, question.ID,
+		)
+
+		if err != nil {
+			tx.Rollback()
+			return err
+		}
+	}
+
+	if err := tx.Commit(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Update a question in the database.
+func UpdateQuestion(db *sql.DB, question *Question) error {
+	// Start a transaction
+	ctx := context.Background()
```

Comment:
take ctx (type `context.Context`) as parameter to this function instead.
in your handler you can get the context of a request from `c.Request().Context()`